Yumemi-ios-training/Yumemi-ios-training/Model/WeatherModel.swift

```diff
@@ -0,0 +1,31 @@
+//
+//  WeatherModel.swift
+//  Yumemi-ios-training
+//
+//  Created by 工藤 海斗 on 2022/03/24.
+//
+
+import Foundation
+import YumemiWeather
+
+protocol WeatherModelProtocol {
+    func feachWeaher() -> Result<WeatherType, APIError>
+}
+
+class WeatherModel: WeatherModelProtocol {
+    func feachWeaher() -> Result<WeatherType, APIError> {
```

Comment:
[imo]
今後どのようなインターフェイスを protocol に持たせていくかにもよりますが、 `WeatherModel` は少し抽象的かなと思いました。
fetch しかしないのであれば、

```suggestion
protocol WeatherFetchable {
    func feachWeaher() -> Result<WeatherType, APIError>
}

class WeatherFetcher: WeatherFetchable {
    func feachWeaher() -> Result<WeatherType, APIError> {
```

等、 API Design Guidelines を参考に付けると命名しやすいかもです。
c.f. [Swiftの命名規則を理解する（Swift API Design Guidelines - Naming 日本語まとめ） #capabilityを示すプロトコルには接尾に able, ible, ing をつけよう](https://qiita.com/fuwamaki/items/f2df71723ab277dffc29#capability%E3%82%92%E7%A4%BA%E3%81%99%E3%83%97%E3%83%AD%E3%83%88%E3%82%B3%E3%83%AB%E3%81%AB%E3%81%AF%E6%8E%A5%E5%B0%BE%E3%81%AB-able-ible-ing-%E3%82%92%E3%81%A4%E3%81%91%E3%82%88%E3%81%86)